File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
    var ret;

    if (!input.trim()) {
        ret = 'Fine. Be that way!';
    } else if (input.toUpperCase() === input && input.toLowerCase() != input) {
        ret = 'Whoa, chill out!';
    } else if (input.endsWith('?')) {
        ret = 'Sure.';
    } else {
        ret = 'Whatever.';
    }

    return ret;
};

module.exports = Bob;

File: go/bob/bob.go
package bob

import "strings"

const testVersion = 2

func Hey(q string) (a string) {
	q = strings.TrimSpace(q)

	if len(q) == 0 {
		a = "Fine. Be that way!"
	} else if strings.ToUpper(q) == q && strings.ToLower(q) != strings.ToUpper(q) {
		a = "Whoa, chill out!"
	} else if q[len(q)-1] == '?' {
		a = "Sure."
	} else {
		a = "Whatever."
	}
	return
}
